### PR TITLE
Adds Integration Tests for Search Pipeline

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Adds Integration Tests for Search Pipeline

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
@msfroh @dbwiddis @saratvemulapalli can you take a look?

**Review by dbwiddis — APPROVED** [review]
Yay more tests!

**Review by joshpalis — APPROVED** [review]

File: modules/search-pipeline-common/src/internalClusterTest/java/org/opensearch/search/pipeline/common/SearchPipelineCommonIT.java [hunk 1]
```diff
@@ -8,41 +8,191 @@
...
+    private static final String PIPELINE_NAME = "test_pipeline";
+
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
         return List.of(SearchPipelineCommonModulePlugin.class);
     }
 
+    @Before
+    public void setup() throws Exception {
+        // Create test index
```
Comment (andrross) [inline modules/search-pipeline-common/src/internalClusterTest/java/org/opensearch/search/pipeline/common/SearchPipelineCommonIT.java:55]: You don't need comments like "create test index" or "create pipeline" when the method names are obvious, like `createIndex()` and `createPipeline()`

File: modules/search-pipeline-common/src/internalClusterTest/java/org/opensearch/search/pipeline/common/SearchPipelineCommonIT.java [hunk 2]
```diff
@@ -8,41 +8,191 @@
...
+
     public void testFilterQuery() {
         // Create a pipeline with a filter_query processor.
-        String pipelineName = "foo";
+        createPipeline();
+
+        // Search without the pipeline. Should see both documents.
+        SearchRequest req = new SearchRequest(TEST_INDEX).source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()));
+        SearchResponse rsp = client().search(req).actionGet();
+        assertEquals(2, Objects.requireNonNull(rsp.getHits().getTotalHits()).value);
```
Comment (andrross) [inline modules/search-pipeline-common/src/internalClusterTest/java/org/opensearch/search/pipeline/common/SearchPipelineCommonIT.java:84]: I'd remove the noise of `Objects.requireNonNull`. Just let the code NPE or fail the equality check if null is returned anywhere.

File: modules/search-pipeline-common/src/internalClusterTest/java/org/opensearch/search/pipeline/common/SearchPipelineCommonIT.java [hunk 3]
```diff
@@ -62,35 +212,11 @@ public void testFilterQuery() {
...
-        rsp = client().search(req).actionGet();
-        assertEquals(1, rsp.getHits().getTotalHits().value);
-
-        // Clean up.
-        ackRsp = client().admin().cluster().deleteSearchPipeline(new DeleteSearchPipelineRequest(pipelineName)).actionGet();
-        assertTrue(ackRsp.isAcknowledged());
-        ackRsp = client().admin().indices().delete(new DeleteIndexRequest(indexName)).actionGet();
+    private void deletePipeline() {
+        AcknowledgedResponse ackRsp;
+        ackRsp = client().admin().cluster().deleteSearchPipeline(new DeleteSearchPipelineRequest(PIPELINE_NAME)).actionGet();
```
Comment (andrross) [inline modules/search-pipeline-common/src/internalClusterTest/java/org/opensearch/search/pipeline/common/SearchPipelineCommonIT.java:219]:
```suggestion
        AcknowledgedResponse ackRsp = client().admin().cluster().deleteSearchPipeline(new DeleteSearchPipelineRequest(PIPELINE_NAME)).actionGet();
```

**Review by andrross — COMMENTED** [review]
Thanks for adding some tests! Apologies in advance for the nitpicking :)

**Review by msfroh — COMMENTED** [review]
These tests are flexing core functionality (default pipelines, update configuration, etc). Do they belong in the integ tests for the `search-pipeline-common` module? Of course, testing that stuff from core is tricky, because there are no processors registered in core.

As @dbwiddis said, "Yay more tests!" -- while I'd love to see core functionality tested from core, I'm also happy to see integ tests for core functionality in this module (since that's a lot better than not having integ tests anywhere).

**Comment by owaiskazi19** [comment]
> These tests are flexing core functionality (default pipelines, update configuration, etc). Do they belong in the integ tests for the `search-pipeline-common` module? Of course, testing that stuff from core is tricky, because there are no processors registered in core.

Do you think this is not the right place to add these tests? The reason you mentioned above was the sole reason I added it in `search-pipeline-common`.

**Review by msfroh — APPROVED** [review]

**Pull request merged**